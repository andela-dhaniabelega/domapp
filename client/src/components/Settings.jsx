import React from 'react';
import UpdateProfileForm from './UpdateProfileForm';
import UpdatePasswordForm from './UpdatePasswordForm';

/**
 * @class Documents
 * @extends {React.Component}
 */
const Settings = () => ({
	/**
	 * @description Render content to the screen
	 * @return {void}
	 */
	render() {
		return (
			<div className="settings-panel f-center">
				<h5 className="settings-panel-header f-center"><span>Settings</span></h5>
				<div className="row">
					<div className="col m6">
						<div className="col s4 l12 darken-1">
							<div className="card settings-card">
								<div className="card-content">
									<span className="settings-title">Profile Details</span>
									<UpdateProfileForm />
								</div>
							</div>
						</div>
					</div>
					<div className="col m6">
						<div className="col s4 l12 darken-1">
							<div className="card settings-card">
								<div className="card-content">
									<span className="settings-title">Change Password</span>
									<UpdatePasswordForm />
								</div>
							</div>
						</div>
					</div>
				</div>
			</div>
		);
	}
});
export default Settings;

