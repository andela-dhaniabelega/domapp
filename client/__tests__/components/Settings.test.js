import React from 'react';
import { shallow } from 'enzyme';
import Settings from '../../src/components/Settings';

describe('Footer', () => {
	it('renders a snapshot', () => {
		const wrapper = shallow(<Settings />);
		expect(wrapper).toMatchSnapshot();
	});
});
