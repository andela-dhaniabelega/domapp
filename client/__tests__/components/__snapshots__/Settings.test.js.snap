// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="settings-panel f-center"
>
    <h5
        className="settings-panel-header f-center"
    >
        <span>
            Settings
        </span>
    </h5>
    <div
        className="row"
    >
        <div
            className="col m6"
        >
            <div
                className="col s4 l12 darken-1"
            >
                <div
                    className="card settings-card"
                >
                    <div
                        className="card-content"
                    >
                        <span
                            className="settings-title"
                        >
                            Profile Details
                        </span>
                        <withRouter(Connect(UpdateProfileForm)) />
                    </div>
                </div>
            </div>
        </div>
        <div
            className="col m6"
        >
            <div
                className="col s4 l12 darken-1"
            >
                <div
                    className="card settings-card"
                >
                    <div
                        className="card-content"
                    >
                        <span
                            className="settings-title"
                        >
                            Change Password
                        </span>
                        <withRouter(Connect(UpdatePasswordForm)) />
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="settings-panel f-center"
>
      <h5
            className="settings-panel-header f-center"
      >
            <span>
                  Settings
            </span>
      </h5>
      <div
            className="row"
      >
            <div
                  className="col m6"
            >
                  <div
                        className="col s4 l12 darken-1"
                  >
                        <div
                              className="card settings-card"
                        >
                              <div
                                    className="card-content"
                              >
                                    <span
                                          className="settings-title"
                                    >
                                          Profile Details
                                    </span>
                                    <withRouter(Connect(UpdateProfileForm)) />
                              </div>
                        </div>
                  </div>
            </div>
            <div
                  className="col m6"
            >
                  <div
                        className="col s4 l12 darken-1"
                  >
                        <div
                              className="card settings-card"
                        >
                              <div
                                    className="card-content"
                              >
                                    <span
                                          className="settings-title"
                                    >
                                          Change Password
                                    </span>
                                    <withRouter(Connect(UpdatePasswordForm)) />
                              </div>
                        </div>
                  </div>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Settings />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Object {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "render": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="settings-panel f-center"
>
          <h5
                    className="settings-panel-header f-center"
          >
                    <span>
                              Settings
                    </span>
          </h5>
          <div
                    className="row"
          >
                    <div
                              className="col m6"
                    >
                              <div
                                        className="col s4 l12 darken-1"
                              >
                                        <div
                                                  className="card settings-card"
                                        >
                                                  <div
                                                            className="card-content"
                                                  >
                                                            <span
                                                                      className="settings-title"
                                                            >
                                                                      Profile Details
                                                            </span>
                                                            <withRouter(Connect(UpdateProfileForm)) />
                                                  </div>
                                        </div>
                              </div>
                    </div>
                    <div
                              className="col m6"
                    >
                              <div
                                        className="col s4 l12 darken-1"
                              >
                                        <div
                                                  className="card settings-card"
                                        >
                                                  <div
                                                            className="card-content"
                                                  >
                                                            <span
                                                                      className="settings-title"
                                                            >
                                                                      Change Password
                                                            </span>
                                                            <withRouter(Connect(UpdatePasswordForm)) />
                                                  </div>
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="settings-panel f-center"
>
          <h5
                    className="settings-panel-header f-center"
          >
                    <span>
                              Settings
                    </span>
          </h5>
          <div
                    className="row"
          >
                    <div
                              className="col m6"
                    >
                              <div
                                        className="col s4 l12 darken-1"
                              >
                                        <div
                                                  className="card settings-card"
                                        >
                                                  <div
                                                            className="card-content"
                                                  >
                                                            <span
                                                                      className="settings-title"
                                                            >
                                                                      Profile Details
                                                            </span>
                                                            <withRouter(Connect(UpdateProfileForm)) />
                                                  </div>
                                        </div>
                              </div>
                    </div>
                    <div
                              className="col m6"
                    >
                              <div
                                        className="col s4 l12 darken-1"
                              >
                                        <div
                                                  className="card settings-card"
                                        >
                                                  <div
                                                            className="card-content"
                                                  >
                                                            <span
                                                                      className="settings-title"
                                                            >
                                                                      Change Password
                                                            </span>
                                                            <withRouter(Connect(UpdatePasswordForm)) />
                                                  </div>
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Settings />,
}
`;
